app/models/stream_events/classification_event.rb

```diff
@@ -11,7 +11,15 @@ def initialize(stream, hash)
     def process
       return unless enabled?
       cache_linked_models!
-      stream.queue.add(ExtractWorker, classification.id) unless classification.workflow.paused?
+
+      workflow = classification.workflow
+
+      return if workflow.paused? || workflow.halted? || workflow.extractors.blank?
+
+      worker = ExtractWorker
+      worker = ExtractWorkerExternal if workflow.has_external_extractors?
```

Comment:
can this just be a standard if block?
```suggestion
      worker = 
        if workflow.has_external_extractors?
          ExtractWorkerExternal
        else 
          ExtractWorker
        end  
```